### PR TITLE
textql 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by dinedal**
Textql allows fast execution of SQL statements over text files. Supports a *nix-like interface, stdin/stdout and pipes. Avoids disk access unless requested.

### CODE REVIEW

File: Library/Formula/textql.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    url "https://github.com/mattn/go-sqlite3.git",
+      :revision => "8897bf145272af4dd0305518cfb725a5b6d0541c"
+  end
+
+  def install
+    (buildpath + "src/github.com/dinedal/textql").install "inputs", "outputs", "storage", "sqlparser", "util", "cmd"
+    ENV["GOPATH"] = buildpath
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-i", "-ldflags", "-X main.VERSION 2.0.0", "#{buildpath}/src/github.com/dinedal/textql/cmd/textql.go"
```
Comment (bfontaine) [inline Library/Formula/textql.rb]: Why is `-i` necessary here?

File: Library/Formula/textql.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    bin.install "textql"
+  end
+
+  test do
+    version = pipe_output("#{bin}/textql --version")
+    assert_match /2.0.0/, version
+    word_count_wc = pipe_output("wc -l /usr/share/dict/words | cut -d\" \" -f 3")
+    word_count_textql = pipe_output("#{bin}/textql -sql 'select count(*) from words' /usr/share/dict/words")
```
Comment (bfontaine) [inline Library/Formula/textql.rb]: Please create a dummy file here so we don’t rely on `/usr/share/dict/words` being present. Something like the following should work:

```rb
(testpath/"test").write "a\nb\nc\n"
assert_match "3\n", shell_output("#{bin}/textql -sql 'select count(*) from test' #{testpath}/test")
```

Comment (bfontaine) [inline Library/Formula/textql.rb]: If `textql` supports reading from stdin we could even shorten this to something like:

```rb
assert_equal "3\n", pipe_output("#{bin}/textql -sql 'select count(*) from ???'", "a\nb\nc\n")
```

**Comment by dinedal** [comment]
Thank you for the feedback, all of your suggestions / corrections have been applied. Please let me know if there is anything else, and I'm happy to adjust it.

**Comment by bfontaine** [comment]
The test failed:
```
==> /usr/local/Cellar/textql/2.0.0/bin/textql -sql 'select count(*) from stdin'
2015/10/11 12:39:33 no such table: stdin
```

**Comment by bfontaine** [comment]
Ping?

**Comment by dinedal** [comment]
Really sorry, I released a new version to fix the issue, but didn't get to make the changes here.

Should I re-open with a new, updated branch?

**Comment by bfontaine** [comment]
No, you can update this PR by either pushing a new commit or squashing in the existing one and force-pushing.

**Comment by dinedal** [comment]
Force pushed a new commit.

File: Library/Formula/textql.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def install
+    (buildpath + "src/github.com/dinedal/textql").install "inputs", "outputs", "storage", "sqlparser", "util", "textql"
+    ENV["GOPATH"] = buildpath
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-ldflags", "-X main.VERSION=2.0.2", "#{buildpath}/src/github.com/dinedal/textql/textql/main.go"
+    mv "main", "textql"
+    bin.install "textql"
```
Comment (bfontaine) [inline Library/Formula/textql.rb]: You can merge the three lines above by adding `"-o", "#{bin}/textql"` to the `go build` line.

File: Library/Formula/textql.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  test do
+    version = pipe_output("#{bin}/textql --version")
+    assert_match /2.0.2/, version
```
Comment (bfontaine) [inline Library/Formula/textql.rb]: You can remove the version test since there’s a better test below.

**Comment by dinedal** [comment]
Changes made as requested, happy to do any more you feel are needed.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @dinedal; we appreciate it! :tada:
